### PR TITLE
5431 torch add113 drop17 baseimage2210

### LINKED ISSUE #5431: tests for torch 1.13 (and drop support for torch 1.7.x?)
**Is your feature request related to a problem? Please describe.**
- add tests for the new pytorch releases
- drop the tests for v1.7.x?

### LINKED ISSUE #5452: Upgrade PyTorch base docker to 22.10
**Is your feature request related to a problem? Please describe.**
As the new PyTorch docker had been released: https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch.
It would be nice to update our dependency and CI tests to cover it.

### PR DESCRIPTION
**Pull request by wyli**
Fixes #5431 
Fixes #5452

### Description
- deprecating pytorch 1.7 tests
- adds pytorch 1.13 tests
- upgrade base image to pytorch 22.10

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by wyli** [comment]
seems to work fine with the github runner, would be great to test the same environments for blossom @YanxuanLiu before you approve this PR

File: .github/workflows/pythonapp-gpu.yml [hunk 1]
```diff
@@ -20,46 +21,35 @@ jobs:
...
             base: "nvcr.io/nvidia/pytorch:21.10-py3"
+          - environment: PT110+CUDA111
+            pytorch: "torch==1.10.2 torchvision==0.11.3 --extra-index-url https://download.pytorch.org/whl/cu111"
+            base: "nvcr.io/nvidia/cuda:11.1.1-devel-ubuntu18.04"
           - environment: PT112+CUDA118
-            # we explicitly set pytorch to -h to avoid pip install error
             # 22.09: 1.13.0a0+d0d6b1f
-            pytorch: "-h"
+            pytorch: "-h"  # we explicitly set pytorch to -h to avoid pip install error
             base: "nvcr.io/nvidia/pytorch:22.09-py3"
```
Comment (YanxuanLiu) [inline .github/workflows/pythonapp-gpu.yml:46]: Can we only use pytorch or cuda images. It could be confusing using both.

Comment (wyli) [reply .github/workflows/pythonapp-gpu.yml:45]: thanks Yanxuan, pytorch docker for example `22.09` is based on pytorch `1.13.0a0+d0d6b1f` which is not exactly the same as pip install pytorch==1.12 or 1.13, so it's necessary to test with these base images. I updated the option names to make it clear

**Comment by wyli** [comment]
this is tested on blossom @YanxuanLiu @Nic-Ma I think it's ready

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**